tests/multi-platform/multi-platform.go

```diff
@@ -0,0 +1,236 @@
+package multiplatform
+
+import (
+	"context"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/devfile/library/v2/pkg/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appservice "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	buildservice "github.com/redhat-appstudio/build-service/api/v1alpha1"
+	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+const (
+	Ec2ArmTag           = "multi-platform-e2e-arm64"
+	HostConfig          = "host-config"
+	ControllerNamespace = "multi-platform-controller"
+	SecretName          = "awskeys"
+)
+
+var (
+	testProjectGitUrl   = utils.GetEnv("MULTI_PLATFORM_TEST_REPO_URL", "https://github.com/devfile-samples/devfile-sample-go-basic")
+	testProjectRevision = utils.GetEnv("MULTI_PLATFORM_TEST_REPO_REVISION", "c713067b0e65fb3de50d1f7c457eb51c2ab0dbb0")
+)
+
+var _ = framework.JVMBuildSuiteDescribe("Multi Platform Controller E2E tests", Label("multi-platform-controller"), func() {
+	var f *framework.Framework
+	AfterEach(framework.ReportFailure(&f))
+	var err error
+
+	defer GinkgoRecover()
+
+	var testNamespace, applicationName, componentName string
+	var component *appservice.Component
+	var timeout, interval time.Duration
+
+	AfterAll(func() {
+		if !CurrentSpecReport().Failed() {
+			Expect(f.AsKubeAdmin.HasController.DeleteComponent(componentName, testNamespace, false)).To(Succeed())
+			Expect(f.AsKubeAdmin.HasController.DeleteApplication(applicationName, testNamespace, false)).To(Succeed())
+			Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			Expect(f.SandboxController.DeleteUserSignup(f.UserName)).To(BeTrue())
+		} else {
+			Expect(f.AsKubeAdmin.CommonController.StoreAllPods(testNamespace)).To(Succeed())
+			Expect(f.AsKubeAdmin.TektonController.StoreAllPipelineRuns(testNamespace)).To(Succeed())
+		}
+	})
+
+	BeforeAll(func() {
+
+		f, err = framework.NewFramework(utils.GetGeneratedNamespace("multi-platform-build"))
+		Expect(err).NotTo(HaveOccurred())
+		testNamespace = f.UserNamespace
+		Expect(testNamespace).NotTo(BeNil(), "failed to create sandbox user namespace")
+
+		Expect(err).ShouldNot(HaveOccurred())
+
+		armInstances, err := getAwsInstances()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		hostConfig, err := f.AsKubeAdmin.CommonController.GetConfigMap(HostConfig, ControllerNamespace)
```

Comment:
not seeing this `host-config` configMap in the development cluster? Do we have to add the host-config [here](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/multi-platform-controller/development)?

Reply:
We can just create it in the test, I need to overwrite all the values anyway.

Reply:
creating it in the tests will also work.